torch/nn/utils/meta_init.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from contextlib import contextmanager
+from itertools import chain
+from typing import Iterator, Union
+
+import torch
+from torch import Tensor
+from torch.nn import Module
+
+
+@contextmanager
+def meta_init() -> Iterator[None]:
```

Comment:
What is the reason we want to have a context manager instead of the initial proposal of a function to initialize a module on meta device?

Reply:
This was specifically asked by the PyTorch Lightning folks since otherwise they would require their users to change their code. By wrapping the context manager inside their API, they make meta initialization transparent to users' code.